service/frontend/workflow_handler.go

```diff
@@ -5182,8 +5184,47 @@ func (wh *WorkflowHandler) GetDeploymentReachability(
 func (wh *WorkflowHandler) UpdateWorkflowExecutionOptions(
 	ctx context.Context,
 	request *workflowservice.UpdateWorkflowExecutionOptionsRequest,
-) (*workflowservice.UpdateWorkflowExecutionOptionsResponse, error) {
-	return nil, status.Errorf(codes.Unimplemented, "method UpdateWorkflowExecutionOptions not implemented")
+) (_ *workflowservice.UpdateWorkflowExecutionOptionsResponse, retError error) {
+	defer log.CapturePanic(wh.logger, &retError)
+	wh.logger.Debug("Received UpdateWorkflowExecutionOptions")
+	if request == nil {
+		return nil, errRequestNotSet
+	}
+	if request.GetWorkflowExecution().GetWorkflowId() == "" {
+		return nil, errWorkflowIDNotSet
+	}
+	if request.GetUpdateMask() == nil {
+		return nil, serviceerror.NewInvalidArgument("UpdateMask is required")
+	}
+	opts := request.GetWorkflowExecutionOptions()
+	if opts == nil {
+		return nil, serviceerror.NewInvalidArgument("WorkflowExecutionOptions is required")
+	}
+	_, err := fieldmaskpb.New(opts, request.GetUpdateMask().GetPaths()...) // errors if paths are not valid for WorkflowExecutionOptions
+	if err != nil {
+		return nil, serviceerror.NewInvalidArgument(fmt.Sprintf("error parsing UpdateMask: %s", err.Error()))
+	}
+	if opts.GetVersioningOverride().GetBehavior() == enumspb.VERSIONING_BEHAVIOR_PINNED &&
+		opts.GetVersioningOverride().GetDeployment() == nil {
+		return nil, serviceerror.NewInvalidArgument("Deployment must be set if behavior override is PINNED")
+	}
```

Comment:
May be just set it? There is `common/enums/defaults.go` for it.

Reply:
I wouldn't know what to set the deployment to though? Pinning to an undefined deployment doesn't make sense.

I could have done "if deployment is nil, pin to whatever deployment is currently there" but that's unsafe, since the base deployment could change and be something the user does not intend, so I want people to intentionally choose a deployment if they're setting a pinned override

Reply:
Did I understand your comment correctly here?

Reply:
sorry, it was late when I left this comment. it doesn't make sense :-)

Reply:
ok, just making sure there is nothing I have to address here :)